presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1535,6 +1538,34 @@ private void analyzeWindowFunctions(QuerySpecification node, List<Expression> ou
 
                 Window window = windowFunction.getWindow().get();
 
+                ImmutableList.Builder<SortItem> orderbyConst = ImmutableList.builder();
+                window.getOrderBy().ifPresent(
+                        orderBy -> orderbyConst.addAll(
+                                orderBy.getSortItems()
+                                .stream()
+                                .filter(item -> item.getSortKey() instanceof Literal).collect(Collectors.toList())));
+
+                List<SortItem> constSortItems = orderbyConst.build();
+                if (!constSortItems.isEmpty()) {
+                    if (isAllowWindowOrderByLiterals(session)) {
+                        warningCollector.add(
+                                new PrestoWarning(
+                                        PERFORMANCE_WARNING,
+                                        String.format(
+                                                "ORDER BY literals/constants with window function: '%s' is unnecessary and expensive. If you intend to ORDER BY using ordinals, please use the actual expression instead: '%s'",
```

Comment:
instead of?

Reply:
Well we already set the context in the first clause. Anyway I fixed it and also removed the unnecessary information.